cpp/ycm/ClangCompleter/TranslationUnit.cpp

```diff
@@ -389,6 +388,68 @@ void TranslationUnit::UpdateLatestDiagnostics() {
   }
 }
 
+namespace {
+  /// Sort a FixIt container by its location's distance from a given column
+  /// (such as the cursor location)
+  ///
+  /// PreCondition: all FixIts in the container are on the same line
+  struct sort_by_location {
+    sort_by_location( int column )
+      : column_(column) {
+
+    }
+
+    bool operator()( const FixIt& a, const FixIt& b ) {
+      int a_distance = a.location.column_number_ - column_;
+      int b_distance = b.location.column_number_ - column_;
+
+      return std::abs(a_distance) < std::abs(b_distance);
+    }
+
+  private:
+    int column_;
+  };
+}
+
+std::vector< FixIt > TranslationUnit::GetFixItsForLocationInFile(
+  int line,
+  int column,
+  const std::vector< UnsavedFile > &unsaved_files,
+  bool reparse ) {
+
+  if ( reparse )
+    ReparseForIndexing( unsaved_files );
+
+  std::vector< FixIt > fixits;
+
+  {
+    unique_lock< mutex > lock( diagnostics_mutex_ );
+
+    for ( std::vector< Diagnostic >::const_iterator it
```

Comment:
Oh God I hate C++98 so much... we need to move the minimum to C++11 some time soon.